### ISSUE TITLE
Make `cidr-matcher` dependency optional

### ISSUE THREAD

**Issue by jcarvalho** [DataDog/dd-trace-js#2346]
The latest versions of the library ([2.15.0](https://github.com/DataDog/dd-trace-js/releases/tag/v2.15.0) and [3.2.0](https://github.com/DataDog/dd-trace-js/releases/tag/v3.2.0)) introduced support for Client IP Parsing and Reporting (https://github.com/DataDog/dd-trace-js/pull/2166).

This change introduced the `cidr-matcher` dependency as a mandatory dependency, which in turn brings the `ip6addr` dependency, which is [Mozilla Public License 2.0](https://github.com/TritonDataCenter/node-ip6addr/blob/master/LICENSE) Licensed. This license is considered a "weak copyleft" license, and as such, it is not allowed by our Open Source License Policy (and I suppose that's the case for many other organizations), forcing us to stay in older versions of the DD Tracer (which don't include the dependency).

Looking at the PR, it seems like the library is only used when running on a version of Node prior to 14.18.0, so I was wondering whether we could make the dependency optional, and document that it must be included when running in older versions of Node?

Alternatively, it seems the [ip-cidr](https://www.npmjs.com/package/ip-cidr) library provides similar functionality, and all its dependencies are MIT and BSD-3 Licensed, so it could be a viable replacement without copyleft licenses. [ipaddr.js](https://www.npmjs.com/package/ipaddr.js) also seems to provide the same functionality, but with a MIT license and no dependencies.

**Comment by simon-id** [comment]
Thank you for reporting this issue with such details!

Just as a note:
- `ip-cidr` cannot be used as it doesn't understand IPv4 mapped over IPv6.
```js
const IpCidr = require('ip-cidr')

const range = new IpCidr('127.0.0.0/8')

range.contains('::ffff:127.0.0.1') // returns false
```
- It looks like `ipaddr.js` is suitable in terms of functionality, but the API is reversed which makes it not optimized for what we want. I will make benchmarks to see what we'll trade off for:
```js
const ipaddrJs = require('ipaddr.js')

const range = ipaddrJs.parseCIDR('127.0.0.0/8')

ipaddrJs.process('::ffff:127.0.0.1').match(range) // both returns true
ipaddrJs.process('::ffff:7f00:0001').match(range)
```
- Once I have the results of the benchmarks we'll be able to make a decision on if we should switch dependency or make the current dep optional.

**Comment by simon-id** [comment]
@jcarvalho Can you please share what tool you use to check all licenses in your dependency tree ? We would like to add it to our CI so this doesn't happen again.

**Comment by jcarvalho** [comment]
@simon-id We use [mend.io](https://www.mend.io/) for License checking.

**Comment by simon-id** [comment]
PR has been merged, it will be shipped in the next release. I'll notify the thread once that happens.

**Comment by jcarvalho** [comment]
Awesome, thanks for the quick turnaround!

**Comment by simon-id** [comment]
@jcarvalho Landed in `dd-trace@3.3.0` please inform us if you're still experiencing issues!

**Comment by jcarvalho** [comment]
Looking good, thanks!
